modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/actions/deleteAssetEntriesBulkAction.ts

```diff
@@ -33,39 +70,109 @@ function getBulkDeleteMessage(selectedData: any) {
 	};
 }
 
-export default function deleteAssetEntriesBulkAction({
-	actionId,
-	selectedData,
-}: {
-	actionId: string;
-	selectedData: any;
-}) {
+/**
+ * Fetches asset library spaces for the given items.
+ */
+async function getEntriesSpaces(items: any[]): Promise<any[]> {
+	const promises = items
+		.filter((item) => item.embedded.scopeId)
+		.map((item) =>
+			fetch(
```

Comment:
Could we use here SpaceService.getSpace()? If not... please move this to that file (SpaceService)